build_ssht_cffi.py

```diff
@@ -1,147 +1,84 @@
-    ssht_path + '/src/c',
-]
+with open(os.path.join(this_dir, "src", "ssht_numba", "src", "ssht.h")) as fl:
+    cdef_str = fl.read()
+    ffi.cdef(cdef_str)
 
-extra_link_args=[
-    "-L" + fftw_path
-]
+fftw_path = os.getenv("FFTW_PATH", None)
 
-# the compiler options from the ssht makefile
-extra_compile_args = [
-    '-std=c99',
-#     '-pedantic',
-    '-Wall',
-    '-O3',
-    '-fopenmp',
-    '-DSSHT_VERSION=\"1.2b1\"',
-    '-DSSHT_BUILD=\"`git rev-parse HEAD`\"',
-    '-fPIC'
-]
+extra_link_args = []
+if fftw_path:
+    extra_link_args.append(["-L" + fftw_path])
 
-# avoid including the ssht_about.c source file, it is not needed here
-# for cffi and produces an error
-
-# the ssht_about.c path is the first element of the sorted list
-ssht_src = sorted(glob.glob(os.path.join(ssht_path, 'src', 'c', '*.c')))[1:]
-
-ffi.set_source("_ssht_cffi",
-"""
-#include "ssht_types.h"
-#include "ssht_error.h"
-#include "ssht_sampling.h"
-#include "ssht_dl.h"
-#include "ssht_core.h"
-#include "ssht_adjoint.h"
-""",
-    sources=ssht_src,
-    libraries=["fftw3"],
-    include_dirs=include_dirs,
-    extra_link_args=extra_link_args,
-    extra_compile_args=extra_compile_args)
 
-def build():
+ssht_src_files = sorted(glob.glob(os.path.join(this_dir, "ssht", "src", "c", "*.c")))
+ssht_src_files = [fl for fl in ssht_src_files if not fl.endswith("ssht_about.c")]
+
+ssht_inc_files = sorted(glob.glob(os.path.join(this_dir, "ssht", "src", "c", "*.h")))
+ssht_inc_files = [fl for fl in ssht_inc_files if not fl.endswith("ssht_about.h")]
 
-    cwd = os.getcwd()
+# with open(os.path.join(this_dir, 'ssht', 'makefile')) as fl:
+#     for line in fl.readlines():
+#         if line.lstrip().startswith("OPT"):
+#             makeopts = line.split("=", 1)[-1].split(" -")
+# makeopts = ["-"+mk for mk in makeopts if (mk and mk != 'pedantic')]
+# makeopts += ['-fPIC']
 
-    ffi.compile(os.path.join(cwd, 'ssht_numba'), verbose=True)
+makeopts = [
+    "-std=c99",
+    "-Wall",
+    "-O3",
+    "-fopenmp",
+    '-DSSHT_VERSION="1.2b1"',
+    '-DSSHT_BUILD="`git rev-parse HEAD`"',
+    "-fPIC",
+]
 
-    files_to_delete = ['_ssht_cffi.o', '_ssht_cffi.c']
-    build_directory = ssht_path.split('/')[1]
+inc_string = ""
+for inc in ssht_inc_files:
+    inc_string += '#include "{}"\n'.format(os.path.basename(inc))
 
-    for file in files_to_delete:
-        # print os.path.join(cwd, file)
-        os.remove(os.path.join(cwd,'ssht_numba', file))
+include_dirs = [os.path.join(this_dir, "ssht", "src", "c")]
 
-    # print os.path.join(cwd, build_directory)
-    shutil.rmtree(os.path.join(cwd, 'ssht_numba', build_directory))
+print(makeopts)
+print(inc_string)
+print(ssht_src_files)
 
-if __name__ == '__main__':
-    build()
+ffi.set_source(
+    "ssht_numba._ssht_cffi",
+    inc_string,
+    sources=ssht_src_files,
+    libraries=["fftw3"],
+    include_dirs=include_dirs,
+    extra_link_args=extra_link_args,
+    extra_compile_args=makeopts,
+)
+
+
+# def build():
```

Comment:
We should remove this function rather than have it commented out.

Reply:
Also agreed and removed.